### PR TITLE
dotenv: fix broken package

### PR DESCRIPTION
**Pull request by cptrodolfox**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by cdepillabout** [comment]
@cptrodolfox Thanks for sending this PR.

However, there are a few things that need to be changed before this can be merged.

Specifically, 2. and 3. from this comment:

https://github.com/NixOS/nixpkgs/pull/72923#issuecomment-551259596

**Comment by cptrodolfox** [comment]
Hi @cdepillabout, thank you for the comment. But, I have one question to fix this package I had to use the newest version from Hackage. The error that caused this package to be broken came from upstream. That, It is why I had to modify configuration-hackage2nix.nix, to update the versions of the dotenv package, the megaparsec package, and the hspec-megaparsec package. If I can not modify this file, should I wait for when the update is ready here?

**Comment by cdepillabout** [comment]
@cptrodolfox I'm not quite sure I follow you.

So `dotenv-0.8.0.2` (the current version in nixpkgs), is completely broken and there is no way to fix it?

Even with your following fix?

> ```nix
> dotenv = addBuildTool super.dotenv pkgs.buildPackages.coreutils;
> ```

**Comment by cptrodolfox** [comment]
Not really, If you put dontCheck it will be fixed I think. Because, the build only fails in the test part.

**Comment by cdepillabout** [comment]
@cptrodolfox Ah okay, I see.

Could you update this PR to just mark `dotenv` as `dontCheck`, and make sure to keep your change where you remove it from the `broken-packages` list?

Please remove the other changes from this PR (updating `pkgs/development/haskell-modules/hackage-packages.nix`, and the `default-package-overrides` list).

**Comment by cptrodolfox** [comment]
Hi @cdepillabout , I added dontCheck to the dotenv package (dotenv 0.8.0.2). I observed that a new package is available in package-packages (dotenv_0_8_0_4). Why did this happen?

File: pkgs/development/haskell-modules/configuration-nix.nix [hunk 1]
```diff
@@ -624,5 +624,7 @@ self: super: builtins.intersectAttrs super {
   # need it during the build itself, too.
   cairo = addBuildTool super.cairo self.buildHaskellPackages.gtk2hs-buildtools;
   pango = disableHardening (addBuildTool super.pango self.buildHaskellPackages.gtk2hs-buildtools) ["fortify"];
-
+  
+  # Tests fails because of missing test file
+  dotenv = dontCheck super.dotenv;
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-nix.nix]: @cptrodolfox This looks good except for one thing: This override should actually be in the configuration-common.nix file.

Once you move it there I will go ahead and merge it in!

Comment (cptrodolfox) [reply pkgs/development/haskell-modules/configuration-nix.nix]: Thanks, I just moved the change.

Comment (cptrodolfox) [reply pkgs/development/haskell-modules/configuration-nix.nix]: One question though, would you mind explaining  me I little bit of the pipeline used to update Hackage package set, or pointing me to any doc. Thanks.

Comment (cdepillabout) [reply pkgs/development/haskell-modules/configuration-nix.nix]: @cptrodolfox I don't think there is any good documentation, but there is a video explaining it somewhat:

https://discourse.nixos.org/t/video-tutorial-how-to-re-generate-the-haskell-package-set-for-nix/4074

Also, peti does this weekly live on twitch.  If you look at past broadcasts, you can watch him do it:

https://www.twitch.tv/peti343/videos

**Comment by cdepillabout** [comment]
> I observed that a new package is available in package-packages (dotenv_0_8_0_4).

I think when a package comes from the LTS package set, hackage2nix is smart enough to also pull in the latest version of a package from hackage and make it available.

File: pkgs/development/haskell-modules/configuration-nix.nix [hunk 2]
```diff
@@ -624,5 +624,5 @@ self: super: builtins.intersectAttrs super {
   # need it during the build itself, too.
   cairo = addBuildTool super.cairo self.buildHaskellPackages.gtk2hs-buildtools;
   pango = disableHardening (addBuildTool super.pango self.buildHaskellPackages.gtk2hs-buildtools) ["fortify"];
-
+
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-nix.nix]: Can you remove this change?

File: pkgs/development/haskell-modules/configuration-nix.nix (hunk 1 above)
Comment (cptrodolfox) [reply pkgs/development/haskell-modules/configuration-nix.nix]: Thanks, I already watch the videos, will see the twitch though.

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 3]
```diff
@@ -275,6 +275,7 @@ self: super: {
   dlist = dontCheck super.dlist;
   docopt = dontCheck super.docopt;                      # http://hydra.cryp.to/build/499172/log/raw
   dom-selector = dontCheck super.dom-selector;          # http://hydra.cryp.to/build/497670/log/raw
+  dotenv = dontCheck super.dotenv;
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-common.nix]: This should be fixed with a more recent version of `dotenv`, right?  Could you add a comment here stating that?

Comment (cptrodolfox) [reply pkgs/development/haskell-modules/configuration-common.nix]: That comment is ok?

Comment (cdepillabout) [reply pkgs/development/haskell-modules/configuration-common.nix]: Thanks, very easy to understand!

**Comment by cdepillabout** [comment]
@cptrodolfox Sorry for going back and forth with you so much on this.

Looking good now, so I will merge when CI finishes.

**Comment by cptrodolfox** [comment]
Don't worry about that, it is a learning experience for me. I have been a user of NixOS for at least two years so I wanted to learn more about nixpkgs and how to contribute to it.

**Comment by cdepillabout** [comment]
> I have been a user of NixOS for at least two years so I wanted to learn more about nixpkgs and how to contribute to it.

Well thanks a lot for taking the time to contribute!  We really appreciate you working with us on fixing stuff like this :-)

**Pull request merged**